src/main/java/org/spongepowered/common/mixin/optimization/world/storage/MapDataMixin_MapOptimization.java

```diff
@@ -178,7 +178,7 @@ public void updateVisiblePlayers(final EntityPlayer player, final ItemStack mapS
                 if (!mixinMapInfo.mapOptimizationBridge$isValid()) {
                     this.mapDecorations.remove(player.getName());
                 } else {
-                    if (this.trackingPosition && mapOptimizationImpl$dimensionField.get(this).equals(player.dimension)) {
+                    if (this.trackingPosition && (((Number)mapOptimizationImpl$dimensionField.get(this)).intValue() == player.dimension)) {
```

Comment:
Couldn't you do this instead, rather than boxing and casting to `Number`?

```suggestion
                    if (this.trackingPosition && mapOptimizationImpl$dimensionField.getInt(this) == player.dimension) {
```

[I say this because of what the javadocs have to say about `Field#getInt(Object)`](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Field.html#getInt-java.lang.Object-), `byte` to `int` is a widening conversion so this should work fine in both cases, and would allow for this to succeed.

(Also, style nitpick - space between the cast operator and the target!)